store/iavl/store.go

```diff
@@ -56,15 +64,39 @@ func LoadStore(db dbm.DB, id types.CommitID, pruning types.PruningOptions, lazyL
 		return nil, err
 	}
 
-	return &Store{tree: tree}, nil
+	lastCommit := types.CommitID{
+		Version: tree.Version(),
+		Hash:    tree.Hash(),
+	}
+
+	store := Store{
+		tree:       tree,
+		pruning:    pruning,
+		version:    tree.Version(),
+		lastCommit: lastCommit,
+	}
+
+	return &store, nil
 }
 
 // UnsafeNewStore returns a reference to a new IAVL Store with a given mutable
 // IAVL tree reference.
 //
 // CONTRACT: The IAVL tree should be fully loaded.
-func UnsafeNewStore(tree *iavl.MutableTree) *Store {
-	return &Store{tree: tree}
+// CONTRACT: PruningOptions passed in as argument must be the same as pruning options
+// passed into iavl.MutableTree
+func UnsafeNewStore(tree *iavl.MutableTree, po types.PruningOptions) *Store {
```

Comment:
This is only used in the tests. I don't feel comfortable with accepting pruning options that could differ from the underlying DB's options. Can we perhaps make this method private?

Reply:
It's used in tests outside the iavl package. I could maybe add a warning to only use in tests

Reply:
Right, I see. In cache and prefix. I think that's fine then as-is then.